Fix entity related issues

wmarques opened the pull request:
Fix #5372 (but i'm not convinced, it's dirty...)
Fix #5347 
Fix #5346

deepu105 commented on generators/entity/templates/client/angular/src/main/webapp/app/entities/_entity-management-detail.component.ts:
```diff
@@ -44,6 +47,7 @@ export class <%= entityAngularName %>DetailComponent implements OnInit, OnDestro
         this.subscription = this.route.params.subscribe(params => {
             this.load(params['id']);
         });
+        this.registerChangeIn<%= entityClassPlural %>();
```
It would be better to do this on `ngOnInit` i think

wmarques replied:
it's already in

deepu105 commented on generators/entity/templates/client/angular/src/main/webapp/app/entities/_entity-management-dialog.component.ts:
```diff
@@ -136,6 +136,8 @@ export class <%= entityAngularName %>DialogComponent implements OnInit {
             this.<%= entityInstance %>Service.create(this.<%= entityInstance %>)
                 .subscribe((res: <%= entityAngularName %>) =>
                     this.onSaveSuccess(res), (res: Response) => this.onSaveError(res.json()));
+                .subscribe((res: <%= entityAngularName %>) => 
```
you have a semicolon before this line

deepu105 replied:
or you should remove the 2 lines before it i guess

wmarques replied:
oh sorry, missrebase 😄

sendilkumarn commented on generators/client/templates/angular/src/main/webapp/app/layouts/main/_main.component.ts:
```diff
@@ -50,6 +50,15 @@ export class <%=jhiPrefixCapitalized%>MainComponent implements OnInit {
                     destinationData = destinationEvent.data;
                     destinationName = destinationEvent.url[0].path;
+                } else {
+                    destinationEvent = event.state.root.children[0];
```
```js 
let destinationEvent = event.state.root.firstChild;
```

&

```js
destinationName = event.url;
```

sendilkumarn replied:
replace the `if-else` by above

wmarques replied:
with this I have an error when accessing to /settings page when not logged.

sendilkumarn replied:
Only for settings or anyother path too?

wmarques replied:
I'll test another path, but I think it's for every protected paths.. I think it's because sometimes our routes have an abstract parent, as admin routes have the admin state that protect them, but others doesn't have any parent, like settings. That's why I did the if else thing above, but I agree, it's not clean.. :disappointed:

sendilkumarn replied:
I tested it against `user-management`

sendilkumarn replied:
Interesting I will have a look this weekend

wmarques replied:
In fact I don't rly know why we need this, isn't the guard sufficient ? Like here https://angular.io/docs/ts/latest/guide/router.html ("Guard the admin feature")

sendilkumarn replied:
Interesting 😕 Is this a bug ?

wmarques replied:
@sendilkumarn I'll try to check this during today or tomorrow, but yes for me we shouldn't need that, the guard should be sufficient.

deepu105 commented on generators/entity/templates/client/angular/src/main/webapp/app/entities/_entity-management-detail.component.ts:
```diff
@@ -8,8 +8,9 @@ for (var idx in fields) {
+import { Subscription } from 'rxjs/Rx';
 <%_ if (enableTranslation) { _%>
-import { JhiLanguageService<% if (fieldsContainBlob) { %>, DataUtils<% } %> } from 'ng-jhipster';
+import { EventManager, JhiLanguageService<% if (fieldsContainBlob) { %>, DataUtils<% } %> } from 'ng-jhipster';
```
No need to add the `<% if (enableTranslation) { %>` as it will be stripped by `copyTemplate`

deepu105 replied:
oh sorry we dont do that for entity files. may be we should

sendilkumarn replied:
Just for one place not worth it

deepu105 replied:
ok

wmarques replied:
It's in another places like entity html template files, we have some jhiTranslate directives, so maybe it's worth, what do u think ?

sendilkumarn replied:
if it is in multiple places, in the same format. we can put that effort. Else, it is okay to be there.

deepu105 replied:
@wmarques the html files are already processed only the js files are not. And I now I remember why it was not done coz of the external ejs templates we have in them as the processJs method cant load multiple external ejs files. so follow @sendilkumarn suggestion here

wmarques commented:
Ok so I checked the "guard issues". We have guards even on non-protected pages like Home or register, that shouldn't be there. So I updated our guard using the "hasAnyAuthority" instead of the "authorize" method and I removed the guard on "open" routes. 

I also removed the whole route-catcher in the main component and it's working pretty well, I'll PR later today or tomorrow to show you that.

deepu105 commented:
@wmarques since this closed quite a few issues I will really like this merged so let me know if you need any help on this?

sendilkumarn commented:
@wmarques any updates on this ?

sendilkumarn commented:
can you fix [this](https://github.com/jhipster/generator-jhipster/pull/5385#discussion_r104852712) one? then merge this straightaway.

wmarques commented:
@sendilkumarn So I need to add an "if (enabletranslation" on JS file that's it ?

wmarques commented:
Ok so I pushed some things but I don't know why router.navigate it's not working when I want to navigate to the stored url...

sendilkumarn commented on generators/client/templates/angular/src/main/webapp/app/shared/auth/_user-route-access-service.ts:
```diff
@@ -1,15 +1,37 @@
+        return this.checkLogin(route.data['authorities'], state.url);
     }
+
+    checkLogin(authorities: string[], url: string): boolean {
```
guess this will always return false, no matter what. 

can you return promise<Boolean>

something like this

```js
return promise.resolve(this.principal.hasAnyAuthority......
```

sendilkumarn commented:
@wmarques ignore [this](https://github.com/jhipster/generator-jhipster/pull/5385#issuecomment-289806996)

sendilkumarn commented:
@wmarques you need any help on this ?

sendilkumarn commented:
I made [this fix](https://github.com/jhipster/generator-jhipster/compare/master...sendilkumarn:fix-entity-crea-error)

jdubois commented:
@wmarques could you have a look before today? I will do a full test tomorrow, and try to release at Devoxx France on Wednesday :-)

sendilkumarn commented:
@jdubois are we going to skip this now ?

jdubois commented:
If it can be merged this morning, then it would be in the next release, and
that would be great.
After that we will be in "code freeze" for the release.

sendilkumarn commented:
Ok I have taken this and rebased and added few fixes on top of it. can we have that ?

jdubois commented:
If you believe it's good (and fix the merge conflict!), don't hesitate to commit -> I will check the master in a couple of hours

sendilkumarn commented:
it just sugar coats the @wmarques' fix and fixes merge conflicts.

wmarques commented:
Sorry guys didn't find time to watch this 😢  Thanks @sendilkumarn for doing this !
